client-js/app/scripts/app.ts

```diff
@@ -247,17 +257,102 @@ window.addEventListener('WebComponentsReady', () => {
     }
   }
 
+  class DatavaultLiveActivity implements Activity {
```

Comment:
I agree there's room here to refactor and clean this up. Happy to take a look at that in a subsequent commit.